tket/src/serialize/pytket/circuit.rs

```diff
@@ -0,0 +1,430 @@
+    /// [`PytketDecodeErrorInner`][super::error::PytketDecodeErrorInner] for
+    /// more details.
+    pub fn reassemble_inline(
+        &self,
+        hugr: &mut Hugr,
+        config: Option<Arc<PytketDecoderConfig>>,
+    ) -> Result<Vec<hugr::Node>, PytketDecodeError> {
+        let options = match &config {
+            Some(config) => DecodeOptions::new().with_config(config.clone()),
+            None => DecodeOptions::new().with_default_config(),
+        };
+
+        for (&original_region, encoded) in &self.circuits {
+            // Decode the circuit into a temporary function node.
+            let Some(signature) = hugr.get_optype(original_region).inner_function_type() else {
+                return Err(PytketDecodeErrorInner::IncompatibleTargetRegion {
+                    region: original_region,
+                    new_optype: hugr.get_optype(original_region).clone(),
+                }
+                .wrap());
+            };
+            let options = options
+                .clone()
+                .with_signature(signature.into_owned())
+                .with_input_params(encoded.input_params.iter().cloned());
+
+            // Run the decoder, generating a new function with the extracted definition.
+            //
+            // Unsupported subgraphs of the original region will be transplanted here.
+            let mut decoder = PytketDecoderContext::new(
+                &encoded.serial_circuit,
+                hugr,
+                DecodeInsertionTarget::Function { fn_name: None },
+                options,
+            )?;
+            decoder.register_opaque_subgraphs(&self.opaque_subgraphs);
+            decoder.run_decoder(&encoded.serial_circuit.commands)?;
+            let decoded_node = decoder.finish()?.node();
+
+            // Replace the region with the decoded function.
+            //
+            // All descendant nodes that were re-used by the decoded circuit got
+            // re-parented at this point, so we can just do a full clear here.
+            while let Some(child) = hugr.first_child(original_region) {
+                hugr.remove_subtree(child);
+            }
+            while let Some(child) = hugr.first_child(decoded_node) {
+                hugr.set_parent(child, original_region);
+            }
+            hugr.remove_node(decoded_node);
+        }
+        Ok(self.circuits.keys().copied().collect_vec())
+    }
+}
+
+impl<Node: HugrNode> EncodedCircuit<Node> {
+    /// Encode a HugrView into a [`EncodedCircuit`].
+    ///
+    /// The HUGR's entrypoint must be a dataflow region that will be encoded as
+    /// the main circuit. Additional circuits may be encoded if
+    /// [`EncodeOptions::encode_subcircuits`] is set.
+    ///
+    /// The circuit may contain opaque barriers encoding opaque subgraphs in the
+    /// original HUGR. These are encoded completely as Hugr envelopes in the
+    /// barrier operations metadata.
```

Comment:
```suggestion
    /// barrier operations' metadata.
```
(disambiguating from `operation's` i.e. being clear, the metadata belongs to multiple operations)